frontend/packages/dev-console/src/components/add/AddCardSection.tsx

```diff
@@ -0,0 +1,104 @@
+import { Switch, Tooltip } from '@patternfly/react-core';
+import { PageLayout } from '@console/shared';
+import { useUserSettings } from '@console/shared/src/hooks/useUserSettings';
+import { AddActionGroup, ResolvedExtension, AddAction } from '@console/dynamic-plugin-sdk';
+import {
+  filterNamespaceScopedUrl,
+  getActionGroups,
+  getSortedExtensionItems,
+} from '../../utils/add-page-utils';
+import { useAccessFilterExtensions } from '../../hooks/hooks';
+import { AddGroup } from '../types';
+import AddCard from './AddCard';
+import useContainerWidth from './useContainerWidth';
+import { MasonryLayout } from './MasonryLayout';
+import './AddPage.scss';
+
+type AddCardsSectionProps = {
+  title: string;
+  namespace: string;
+  hintBlock?: React.ReactNode;
+  addActionExtensions: ResolvedExtension<AddAction>[];
+  addActionGroupExtensions: ResolvedExtension<AddActionGroup>[];
+};
+
+const AddCardsSection: React.FC<AddCardsSectionProps> = ({
+  title,
+  namespace,
+  hintBlock,
+  addActionExtensions,
+  addActionGroupExtensions,
+}) => {
+  const { t } = useTranslation();
+  const [showDetails, setShowDetails] = useUserSettings(
+    'console.addPage.showDetails',
+    'show',
+    true,
+  );
+  const filteredAddActionExtensions: ResolvedExtension<AddAction>[] = filterNamespaceScopedUrl(
+    namespace,
+    useAccessFilterExtensions(namespace, addActionExtensions),
+  );
+
+  const sortedActionGroup: ResolvedExtension<AddActionGroup>[] = getSortedExtensionItems(
+    addActionGroupExtensions,
+  );
+
+  const addGroups: AddGroup[] = getActionGroups(
+    filteredAddActionExtensions.filter(
+      (filterActionGroup) => filterActionGroup.properties.id !== 'import-from-samples',
+    ),
+    sortedActionGroup,
+  );
+
+  const addCards: React.ReactElement[] = addGroups.map(({ id, name, items }) => (
+    <AddCard key={id} title={name} items={items} namespace={namespace} />
+  ));
+
+  const defaultHintText = t(
+    'devconsole~Select a way to create an Application, component or service from one of the options.',
+  );
+
+  const defaultHintBlock: React.ReactNode = (
+    <div className="odc-add-page__hint-block">
+      <div className="odc-add-page__hint-block-text">{defaultHintText}</div>
+      <div className="odc-add-page__hint-block-details-switch">
+        <Tooltip
+          content={
+            <div className="odc-add-page__hint-block-details-switch-tooltip">
+              {t('devconsole~Show or hide details about each item')}
+            </div>
+          }
+          position="top"
+        >
+          <Switch
+            aria-label="Show add card details"
+            isChecked={showDetails === 'show'}
+            onChange={(checked) => {
+              setShowDetails(checked ? 'show' : 'hide');
+            }}
+            data-test-id="add-page-details-switch"
+          />
+        </Tooltip>
+        <span className="odc-add-page__hint-block-details-switch-text">
+          {t('devconsole~Details on')}
```

Comment:
Please show "Details off" if the switch is disabled. See "User has hidden details" in design doc:

```suggestion
          {showDetails === 'show' ? t('devconsole~Details off') : t('devconsole~Details on')}
```